spark/v3.2/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/RewriteDeleteFromTable.scala

```diff
@@ -49,25 +51,42 @@ import org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation
 object RewriteDeleteFromTable extends RewriteRowLevelCommand {
 
   override def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
-    case d @ DeleteFromIcebergTable(aliasedTable, Some(cond), None) if d.resolved =>
+    case d @ DeleteFromIcebergTable(aliasedTable, Some(_), None) if d.resolved =>
       EliminateSubqueryAliases(aliasedTable) match {
         case r @ DataSourceV2Relation(tbl: SupportsRowLevelOperations, _, _, _, _) =>
-          val operation = buildRowLevelOperation(tbl, DELETE)
-          val table = RowLevelOperationTable(tbl, operation)
-          val rewritePlan = operation match {
-            case _: SupportsDelta =>
-              buildWriteDeltaPlan(r, table, cond)
-            case _ =>
-              buildReplaceDataPlan(r, table, cond)
+          rewriteDeleteFromTable(d, r, tbl)
+        case v: View =>
+          val viewChild = v.child
+          val icebergTableView = v.children.size == 1 && viewChild.isInstanceOf[DataSourceV2Relation] &&
```

Comment:
Do we have to use `View`? Do you know what changed?

I’m by no means a catalyst expert, but this seems like an abuse of `View`. Maybe I’m mistaken and this is not uncommon pattern.

Reply:
the major reason to use view is that we have no ways to refer to a table created with HadoopTables.create()..which is usually a pure directory path.... so we have to load the table and then register it as a view, and then we can refer it with the view name in the statement like MERGE INTO, etc.